src/pages/StoryEdit.tsx

```diff
@@ -1,11 +1,24 @@
 import styled from '@emotion/styled';
+import { CircularProgress } from '@mui/material';
+import { useParams } from 'react-router-dom';
 
 import StoryEditForm from '../components/StoryEdit/StoryEditForm';
+import useFetchStory from '../hooks/useFetchStory';
 
 const StoryEdit = () => {
+  const { storyId } = useParams();
+  let title;
+  if (!storyId) {
+    title = '스토리 추가';
+  } else {
+    const { story, isLoading } = useFetchStory();
+    if (isLoading || !story.title) return <CircularProgress />;
+    // TODO: 스토리 정보 Form으로 전달(hook 매개변수로 initialState 전달)
+  }
```

Comment:
제가 알기론 훅은 컴포넌트 함수 내에서 최상위에 선언되어야 하는 것으로 알아요. 커스텀 훅도 해당 규칙이 동일하게 적용되는지는 모르겠는데 혹시 개발하면서 Hook rules 관련된 에러는 발생 안했었나요? 궁금하네요!

그리고 useFetchStory에서도 useParams가 있네요. StoryEdit 컴포넌트에서 storyId를 검사하는 게 아니라 useFetchStory에서도 useParams를 사용해서 storyId를 가져오니, useFetchStory에서 아예 storyId가 없으면 title: '스토리 추가'를 반환값에 넣어서 반환하는 방향은 어떨까요? useParams도 useFetchStory로 빼버릴 수 있을 것 같기도 하구요.

```suggestion
const { story, isLoading, title } = useFetchStory();

if (isLoading || !story.title) return <CircularProgress />;

return (
  <Container>
    <h1>스토리 추가</h1>
    <h1>{title}</h1>
    <StoryEditForm />
  </Container>
);
```

제가 모르는 유리님의 의도가 있는지 궁금합니다.

reference : https://ko.reactjs.org/docs/hooks-rules.html

Reply:
@metacode22 Hook rules 관련 에러는 발생하지 않더라구요..! 

스토리 수정 기능을 다음 차례에 구현하려고 생각하고 고민 없이 작성했는데, 좀 더 면밀하게 살펴보지 못한 것 같습니다;) 
구현 방향에 대해 고민돼서 이번 PR에는 반영하지 못했는데 잘 설명해 주셔서 감사해요!! 꼭 참고하겠습니다 👍

Reply:
아하 아직 완전히 구현한 게 아니셨군용 제가 괜히 헛다리 짚었네요ㅋㅋㅋㅋ 도움이 되었다면 다행입니다! 고생하셨어요!